litr-demo/src/main/java/com/linkedin/android/litr/demo/data/TransformationPresenter.java

```diff
@@ -54,26 +70,105 @@ public void startTransformation(@NonNull SourceMedia sourceMedia,
+                                       createGlFilters(sourceMedia, overlayTarget));
+        } else {
+            performTrackTransformation(sourceMedia, videoTarget, audioTarget, overlayTarget, transformationState, transformationListener);
         }
+    }
 
-        MediaFormat targetAudioFormat = null;
-        if (audioTarget != null && audioTarget.shouldTranscodeAudio) {
-            targetAudioFormat = new MediaFormat();
-            targetAudioFormat.setString(MediaFormat.KEY_MIME, sourceMedia.audioMimeType);
-            targetAudioFormat.setInteger(MediaFormat.KEY_CHANNEL_COUNT,sourceMedia.audioChannelCount);
-            targetAudioFormat.setInteger(MediaFormat.KEY_SAMPLE_RATE, sourceMedia.audioSamplingRate);
-            targetAudioFormat.setInteger(MediaFormat.KEY_BIT_RATE, Integer.parseInt(audioTarget.targetBitrate) * 1000);
+    public void cancelTransformation(@NonNull String requestId) {
+        mediaTransformer.cancel(requestId);
+    }
+
+    public void play(@Nullable File targetFile) {
+        if (targetFile != null && targetFile.exists()) {
+            Intent playIntent = new Intent(Intent.ACTION_VIEW);
+            Uri videoUri = FileProvider.getUriForFile(context,
+                                                      context.getPackageName() + ".provider",
+                                                      targetFile);
+            playIntent.setDataAndType(videoUri, "video/*");
+            playIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+
+            context.startActivity(playIntent);
+        }
+    }
+
+    private void performTrackTransformation(@NonNull SourceMedia sourceMedia,
+                                            @Nullable VideoTarget videoTarget,
+                                            @Nullable AudioTarget audioTarget,
+                                            @Nullable OverlayTarget overlayTarget,
+                                            @NonNull TransformationState transformationState,
+                                            @NonNull TransformationListener transformationListener) {
+        int trackCount = 0;
+        if (videoTarget != null && videoTarget.shouldKeepTrack) {
+            trackCount++;
+        }
+        if (audioTarget != null && audioTarget.shouldKeepTrack) {
+            trackCount++;
+        }
+        if (trackCount < 1) {
+            throw new IllegalArgumentException("No output tracks!");
+        }
+
+        try {
+            MediaTarget mediaTarget = new MediaMuxerMediaTarget(transformationState.targetFile.getAbsolutePath(),
+                                                                trackCount,
+                                                                sourceMedia.videoRotation,
+                                                                MediaMuxer.OutputFormat.MUXER_OUTPUT_MPEG_4);
+
+            List<TrackTransform> trackTransforms = new ArrayList<>(2);
```

Comment:
new ArrayList<>(2) or new ArrayList<>(trackCount)?

Reply:
2 is a maximum. 2 vs. 1 is not a big hit memory wise, and saves us from doing more complex logic to calculate it.